src/Target.cpp

```diff
@@ -164,19 +234,23 @@ Target calculate_host_target() {
         }
         if ((info2[1] & avx512) == avx512) {
             initial_features.push_back(Target::AVX512);
+            // FIXME: port to family/model -based detection.
             if ((info2[1] & avx512_knl) == avx512_knl) {
                 initial_features.push_back(Target::AVX512_KNL);
             }
+            // FIXME: port to family/model -based detection.
             if ((info2[1] & avx512_skylake) == avx512_skylake) {
                 initial_features.push_back(Target::AVX512_Skylake);
             }
+            // FIXME: port to family/model -based detection.
             if ((info2[1] & avx512_cannonlake) == avx512_cannonlake) {
                 initial_features.push_back(Target::AVX512_Cannonlake);
 
                 const uint32_t avx512vnni = 1U << 11;  // vnni result in ecx
                 const uint32_t avx512bf16 = 1U << 5;   // bf16 result in eax, with cpuid(eax=7, ecx=1)
                 int info3[4];
                 cpuid(info3, 7, 1);
+                // FIXME: port to family/model -based detection.
```

Comment:
Could you elaborate on this comment? Why should we detect vnni instructions using the family/model?

Reply:
These comment are about things like
```
initial_features.push_back(Target::AVX512_SapphireRapids);
```
... which is an inherently fragile approach, it relies on the fact
that there is _currently_ no other CPU that supports these features.

For example, if you know that the CPU only supports AVX2,
that doesn't tell you which CPU that is,
it could be any number of CPU's (haswell,znver1,znver2,znver3).